@glyph updated with a `_Numeric` protocol! I tried setting it as a bound on a `TypeVar` and using that for `minimum`, `maximum`, and `kind`, but mypy complains and I couldn't figure out how to make it happy:

```diff
diff --git a/src/klein/_form.py b/src/klein/_form.py
index da93a33..04de050 100644
--- a/src/klein/_form.py
+++ b/src/klein/_form.py
@@ -56,6 +56,9 @@ class _Numeric(Protocol[_T]):
         ...
 
 
+_N = TypeVar("_N", bound=_Numeric)
+
+
 class CrossSiteRequestForgery(Resource):
     """
     Cross site request forgery detected.  Request aborted.
@@ -273,9 +276,9 @@ class Field:
     @classmethod
     def number(
         cls,
-        minimum: Optional[_Numeric] = None,
-        maximum: Optional[_Numeric] = None,
-        kind: Callable[[str], _Numeric] = float,
+        minimum: Optional[_N] = None,
+        maximum: Optional[_N] = None,
+        kind: Callable[[str], _N] = float,
         **kw: Any,
     ) -> "Field":
         """
```

```
src/klein/_form.py:281:37: error: Incompatible default for argument "kind" (default has type "type[float]", argument has type
"Callable[[str], _N]")  [assignment]
            kind: Callable[[str], _N] = float,
                                        ^~~~~
Found 1 error in 1 file (checked 50 source files)
```

--- a/.coveragerc
+++ b/.coveragerc
@@ -15,3 +15,4 @@ source=
 exclude_lines =
     pragma: no cover
     if TYPE_CHECKING:
+    \s*\.\.\.$

--- a/src/klein/_form.py
+++ b/src/klein/_form.py
@@ -43,7 +43,6 @@ from .interfaces import (
 )
 
 
-_T = TypeVar("_T", contravariant=True)
 _Self = TypeVar("_Self")
 
 

--- a/src/klein/_form.py
+++ b/src/klein/_form.py
@@ -14,6 +14,7 @@ from typing import (
     Sequence,
     TypeVar,
     cast,
+    overload,
 )
 
 import attr
@@ -43,17 +44,21 @@ from .interfaces import (
 
 
 _T = TypeVar("_T", contravariant=True)
+_Self = TypeVar("_Self")
 
 
-class _Numeric(Protocol[_T]):
+class _Numeric(Protocol):
     def __float__(self) -> float:
         ...
 
-    def __lt__(self, other: _T) -> bool:
+    def __lt__(self: _Self, other: _Self) -> bool:
         ...
 
-    def __gt__(self, other: _T) -> bool:
+    def __gt__(self: _Self, other: _Self) -> bool:
         ...
+
+
+_N = TypeVar("_N", bound=_Numeric)
 
 
 class CrossSiteRequestForgery(Resource):
@@ -270,12 +275,46 @@ class Field:
             **kw,
         ).maybeNamed(name)
 
+    @overload
     @classmethod
     def number(
         cls,
-        minimum: Optional[_Numeric] = None,
-        maximum: Optional[_Numeric] = None,
-        kind: Callable[[str], _Numeric] = float,
+        minimum: Optional[float] = None,
+        maximum: Optional[float] = None,
+        kind: Callable[[str], float] = float,
+        **kw: Any,
+    ) -> "Field":
+        ...
+
+    @overload
+    @classmethod
+    def number(
+        cls,
+        minimum: Optional[_N] = None,
+        maximum: Optional[_N] = None,
+        *,
+        kind: Callable[[str], _N],
+        **kw: Any,
+    ) -> "Field":
+        ...
+
+    @overload
+    @classmethod
+    def number(
+        cls,
+        minimum: _N,
+        maximum: _N,
+        kind: Callable[[str], _N],
+        **kw: Any,
+    ) -> "Field":
+        ...
+
+    @classmethod
+    def number(
+        cls,
+        minimum: Optional[_N] = None,
+        maximum: Optional[_N] = None,
+        kind: Callable[[str], _N] = float,  # type:ignore[assignment]
         **kw: Any,
     ) -> "Field":
         """

